test/files/neg/t7187-2.13.scala

```diff
@@ -1,4 +1,6 @@
 class EtaExpandZeroArg {
   def foo() = ""
-  val f: () => Any = foo
+  val f: () => Any = foo   // error (not eta-expanded)
+  val g: () => Any = foo _ // ok
+  val h = foo _            // error (not eta-expanded)
```

Comment:
@adriaanm is the above as expected?

Reply:
That looks like a bug (`foo _` should type check), will include that in my eta-expansion work. Hope to work on that today.

```
scala -Xsource:2.13
Welcome to Scala 2.13.0-20170927-190913-5cf58c5 (Java HotSpot(TM) 64-Bit Server VM, Java 1.8.0_152).
Type in expressions for evaluation. Or try :help.

scala> def foo() = ""
foo: ()String

scala> val h = foo _ 
               ^
       error: type mismatch;
        found   : String
        required: () => ?
```

Reply:
https://github.com/scala/bug/issues/10612